server/remote/msgraph/remote.go

```diff
@@ -37,17 +37,24 @@ func NewRemote(conf *config.Config, logger bot.Logger) remote.Remote {
 }
 
 // MakeClient creates a new client for user-delegated permissions.
-func (r *impl) MakeClient(ctx context.Context, poster bot.Poster, store store.Store, mattermostUserID string, token *oauth2.Token) remote.Client {
-	httpClient := r.NewOAuth2Config().Client(ctx, token)
+func (r *impl) MakeClient(ctx context.Context, token *oauth2.Token, kvstore store.Store, mattermostUserID string, checkUserStatus func() bool, changeUserStatus func(error)) remote.Client {
+	config := r.NewOAuth2Config()
+
+	token, err := store.RefreshAndStoreToken(kvstore, token, config, mattermostUserID)
```

Comment:
Maybe we should have a `MakeUserClient` function that calls `RefreshAndStoreToken`, and have the shared `MakeClient` function be more general purpose. I think the current code will have issues with the `MakeSuperuserClient` function calling this `MakeClient`, since there will never be a refresh occurring in that case.